DOCTYPE / DTD path is invalid in --xunit output (Problems during the conversion into JUnit)

Getting path issues when running in Jenkins.


```shell
[xUnit] [ERROR] - Validation error on input
ERROR: Step ‘Publish xUnit test result report’ aborted due to exception: 
hudson.util.IOException2: There are some problems during the conversion into JUnit reports: 
	at org.jenkinsci.plugins.xunit.service.XUnitTransformer.invoke(XUnitTransformer.java:183)
	at org.jenkinsci.plugins.xunit.service.XUnitTransformer.invoke(XUnitTransformer.java:44)
	at hudson.FilePath.act(FilePath.java:990)
	at hudson.FilePath.act(FilePath.java:968)
	at org.jenkinsci.plugins.xunit.XUnitProcessor.performTests(XUnitProcessor.java:145)
	at org.jenkinsci.plugins.xunit.XUnitProcessor.performXUnit(XUnitProcessor.java:88)
	at org.jenkinsci.plugins.xunit.XUnitPublisher.perform(XUnitPublisher.java:142)
	at org.jenkinsci.plugins.xunit.XUnitPublisher.perform(XUnitPublisher.java:134)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:782)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:723)
	at hudson.model.Build$BuildExecution.post2(Build.java:185)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:668)
	at hudson.model.Run.execute(Run.java:1763)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:410)
Caused by: org.jenkinsci.plugins.xunit.exception.XUnitException: Validation error on input
	at org.jenkinsci.plugins.xunit.service.XUnitValidationService.validateInputFile(XUnitValidationService.java:77)
	at org.jenkinsci.plugins.xunit.service.XUnitTransformer.invoke(XUnitTransformer.java:132)
	... 16 more
Caused by: org.jenkinsci.lib.dtkit.util.validator.ValidationException: Validation error
	at org.jenkinsci.lib.dtkit.util.validator.ValidationService.processValidation(ValidationService.java:136)
	at org.jenkinsci.lib.dtkit.model.AbstractOutputMetric.validate(AbstractOutputMetric.java:62)
	at org.jenkinsci.plugins.xunit.types.JUnitInputMetric.validateInputFile(JUnitInputMetric.java:65)
	at org.jenkinsci.plugins.xunit.service.XUnitValidationService.validateInputFile(XUnitValidationService.java:66)
	... 17 more
Caused by: java.io.FileNotFoundException: /var/lib/jenkins/workspace/PortalDev/testsuites (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:90)
	at sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:188)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.setupCurrentEntity(XMLEntityManager.java:623)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.startEntity(XMLEntityManager.java:1305)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.startDTDEntity(XMLEntityManager.java:1271)
	at com.sun.org.apache.xerces.internal.impl.XMLDTDScannerImpl.setInputSource(XMLDTDScannerImpl.java:263)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$DTDDriver.dispatch(XMLDocumentScannerImpl.java:1167)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$DTDDriver.next(XMLDocumentScannerImpl.java:1050)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:964)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:606)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:118)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:510)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:848)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:777)
	at com.sun.org.apache.xerces.internal.jaxp.validation.StreamValidatorHelper.validate(StreamValidatorHelper.java:155)
	at com.sun.org.apache.xerces.internal.jaxp.validation.ValidatorImpl.validate(ValidatorImpl.java:116)
	at javax.xml.validation.Validator.validate(Validator.java:124)
	at org.jenkinsci.lib.dtkit.util.validator.ValidationService.processValidation(ValidationService.java:124)
	... 20 more
```

See https://github.com/n1k0/casperjs/issues/1458#issuecomment-209021781 and on wards for more info.

> Caused by: java.io.FileNotFoundException: /var/lib/jenkins/workspace/PortalDev/testsuites (No such file or directory)

@amerikan can you give us a diff of a xunit report OK / NOK ? this shouldn't reveal on which project your works but probably show us what is broken :)

Nope, same issue when trying with 1.1.1.
```
+ phantomjs --version
1.9.8
+ casperjs --version
1.1.1


[32;1mResult log stored in log.xml                                                    [0m
Build step 'Execute shell' marked build as failure
[xUnit] [INFO] - Starting to record.
[xUnit] [INFO] - Processing JUnit
[xUnit] [INFO] - [JUnit] - 1 test report file(s) were found with the pattern 'log.xml' relative to '/var/lib/jenkins/workspace/PortalDev' for the testing framework 'JUnit'.
[xUnit] [ERROR] - Validation error on input
ERROR: Step ‘Publish xUnit test result report’ aborted due to exception: 
hudson.util.IOException2: There are some problems during the conversion into JUnit reports: 
	at org.jenkinsci.plugins.xunit.service.XUnitTransformer.invoke(XUnitTransformer.java:183)
	at org.jenkinsci.plugins.xunit.service.XUnitTransformer.invoke(XUnitTransformer.java:44)
	at hudson.FilePath.act(FilePath.java:990)
	at hudson.FilePath.act(FilePath.java:968)
	at org.jenkinsci.plugins.xunit.XUnitProcessor.performTests(XUnitProcessor.java:145)
	at org.jenkinsci.plugins.xunit.XUnitProcessor.performXUnit(XUnitProcessor.java:88)
	at org.jenkinsci.plugins.xunit.XUnitPublisher.perform(XUnitPublisher.java:142)
	at org.jenkinsci.plugins.xunit.XUnitPublisher.perform(XUnitPublisher.java:134)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:782)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:723)
	at hudson.model.Build$BuildExecution.post2(Build.java:185)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:668)
	at hudson.model.Run.execute(Run.java:1763)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:410)
Caused by: org.jenkinsci.plugins.xunit.exception.XUnitException: Validation error on input
	at org.jenkinsci.plugins.xunit.service.XUnitValidationService.validateInputFile(XUnitValidationService.java:77)
	at org.jenkinsci.plugins.xunit.service.XUnitTransformer.invoke(XUnitTransformer.java:132)
	... 16 more
Caused by: org.jenkinsci.lib.dtkit.util.validator.ValidationException: Validation error
	at org.jenkinsci.lib.dtkit.util.validator.ValidationService.processValidation(ValidationService.java:136)
	at org.jenkinsci.lib.dtkit.model.AbstractOutputMetric.validate(AbstractOutputMetric.java:62)
	at org.jenkinsci.plugins.xunit.types.JUnitInputMetric.validateInputFile(JUnitInputMetric.java:65)
	at org.jenkinsci.plugins.xunit.service.XUnitValidationService.validateInputFile(XUnitValidationService.java:66)
	... 17 more
Caused by: java.io.FileNotFoundException: /var/lib/jenkins/workspace/PortalDev/testsuites (No such file or directory)
	at java.io.FileInputStream.open0(Native Method)
	at java.io.FileInputStream.open(FileInputStream.java:195)
	at java.io.FileInputStream.<init>(FileInputStream.java:138)
	at java.io.FileInputStream.<init>(FileInputStream.java:93)
	at sun.net.www.protocol.file.FileURLConnection.connect(FileURLConnection.java:90)
	at sun.net.www.protocol.file.FileURLConnection.getInputStream(FileURLConnection.java:188)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.setupCurrentEntity(XMLEntityManager.java:623)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.startEntity(XMLEntityManager.java:1305)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.startDTDEntity(XMLEntityManager.java:1271)
	at com.sun.org.apache.xerces.internal.impl.XMLDTDScannerImpl.setInputSource(XMLDTDScannerImpl.java:263)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$DTDDriver.dispatch(XMLDocumentScannerImpl.java:1167)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$DTDDriver.next(XMLDocumentScannerImpl.java:1050)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl$PrologDriver.next(XMLDocumentScannerImpl.java:964)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(XMLDocumentScannerImpl.java:606)
	at com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(XMLNSDocumentScannerImpl.java:118)
	at com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(XMLDocumentFragmentScannerImpl.java:510)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:848)
	at com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(XML11Configuration.java:777)
	at com.sun.org.apache.xerces.internal.jaxp.validation.StreamValidatorHelper.validate(StreamValidatorHelper.java:155)
	at com.sun.org.apache.xerces.internal.jaxp.validation.ValidatorImpl.validate(ValidatorImpl.java:116)
	at javax.xml.validation.Validator.validate(Validator.java:124)
	at org.jenkinsci.lib.dtkit.util.validator.ValidationService.processValidation(ValidationService.java:124)
	... 20 more
```

--- a/modules/xunit.js
+++ b/modules/xunit.js
@@ -205,9 +205,8 @@ XUnitExporter.prototype.setResults = function setResults(results) {
  * @return void
  */
 XUnitExporter.prototype.setupDocument = function() {
-    // Since we are outputting XML, let's do our own document type
-    var documentType = document.implementation.createDocumentType("casperjs", "-//CasperJS//XUnit Test Results", "testsuites");
-
-    this._xmlDocument = document.implementation.createDocument("", "", documentType);
+    // Note that we do NOT use a documentType here, because validating
+    // parsers try to fetch the (non-existing) DTD and fail #1528
+    this._xmlDocument = document.implementation.createDocument("", "");
     this._xml = this._xmlDocument.appendChild(this._xmlDocument.createElement("testsuites"));
 };

